src/DevYeah.LMS/DevYeah.LMS.Business/CourseService.cs

```diff
@@ -76,6 +76,37 @@ public ServiceResult<CourseServiceResultCode> GetCourseByKey(Guid courseId)
             }
         }
 
+        public ServiceResult<CourseServiceResultCode> GetAllCourses()
+        {
+            try
+            {
+                var allCourses = _courseRepo.GetAllCourses();
+                return BuildResult(true, CourseServiceResultCode.Success, resultObj: allCourses);
+            }
+            catch (Exception ex)
+            {
+                _systemErrorsRepo.AddLog(ex);
+                return InternalErrorResult(CourseServiceResultCode.BackendException);
+            }
+        }
+
+        public ServiceResult<CourseServiceResultCode> GetAllCoursesOfCategory(Guid catId)
+        {
+            if (catId == Guid.Empty) return GetAllCourses();
+            try
+            {
+                var isValidCat = _categoryRepo.IsExisted(catId);
+                if (!isValidCat) return DataErrorResult(CourseServiceResultCode.DataNotExist);
```

Comment:
Alternatively, you can just run query without checking if it exists, then verify it in the UI